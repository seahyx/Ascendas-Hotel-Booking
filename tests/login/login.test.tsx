import React from 'react'
import { render, screen } from "@testing-library/react";
import userEvent from '@testing-library/user-event'
import LoginPage from "../../src/pages/login";

describe("Home", () => {
    const { container } = render(<LoginPage />);
    it("renders correctly", () => {
        expect(container).toMatchSnapshot();
    });

    it("has username & password", () => {
        const { container } = render(<LoginPage />);
        expect(screen.getByText("email:")).toBeInTheDocument;
        expect(screen.getByText("Password:")).toBeInTheDocument;
    });

    it("has button for log in", () => {
        const { container } = render(<LoginPage />);
        userEvent.click(screen.getByRole("button"))
        expect(screen.getByRole('button')).toBeCalled;
    });

    it("allows log in", () => {
        const { container } = render(<LoginPage />);
        userEvent.type(screen.getByText("email:"),"abc@example.com")
        userEvent.type(screen.getByText("Password:"),"password")
        userEvent.click(screen.getByRole("button"))
    });

});