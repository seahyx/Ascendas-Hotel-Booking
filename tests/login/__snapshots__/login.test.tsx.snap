// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home renders correctly 1`] = `
<div>
  <div
    class="container"
  >
    <form>
      <h2>
        Login
      </h2>
      <label
        for="email"
      >
        email:
      </label>
      <input
        id="email"
        name="email"
        required=""
        type="text"
      />
      <label
        for="password"
      >
        Password:
      </label>
      <input
        id="password"
        name="password"
        required=""
        type="password"
      />
      <button
        type="submit"
      >
        Submit
      </button>
    </form>
    <p>
      Not registered yet? 
      <a
        href="/register"
      >
        Register here
      </a>
    </p>
  </div>
</div>
`;
